src/oc/hocdec.h

```diff
@@ -140,17 +143,85 @@ struct Symbol { /* symbol table entry */
 
 using hoc_List = hoc_Item;
 
-union Datum { /* interpreter stack type */
-    double val;
-    Symbol* sym;
-    int i;
-    double* pval; /* first used with Eion in NEURON */
-    Object** pobj;
-    Object* obj; /* sections keep this to construct a name */
-    char** pstr;
-    hoc_Item* itm;
-    hoc_List* lst;
-    void* _pvoid; /* not used on stack, see nrnoc/point.cpp */
+namespace neuron::detail {
+struct _Datum { /* interpreter stack type */
```

Comment:
This name is reserved (underscore followed by a capital letter): https://en.cppreference.com/w/cpp/language/identifiers, and it's not the interpreter stack type any more :)